newsfragments/2539.internal.rst

```diff
@@ -0,0 +1 @@
+Additional contract ``abi`` documentation to make it a clear requirement for contract instances.
```

Comment:
```suggestion
Additional contract ``abi`` documentation to make it a clear requirement for interacting with contract instances.
```

Perhaps that distinction? Are we allowed to create a contract instance without an ABI? I'm not at my computer at the moment to verify but, if so, is there a particular reason we even allow that and don't validate it as an input to `__init__()`?

Just curious on improvements that could come in `v7` but also maybe we should clarify the exceptions in the docs as well, if they exist.

Reply:
I recall an example that used a contract without an abi, I'll dig it up in the docs in a bit. It is possible to create, though very limited in what it can do.

I have no clear idea why it isn't enforced but maybe it is so that really basic operations can be done.maybe comparing contract instances or something?